pypesto/objective/base.py

```diff
@@ -218,23 +241,23 @@ def check_sensi_orders(
             Boolean indicating whether combination of sensi_orders and mode
             is supported
         """
-        raise NotImplementedError()
-
-    @abc.abstractmethod
-    def check_mode(self, mode: str) -> bool:
-        """
-        Check if the objective is able to compute in the requested mode.
-
-        Parameters
-        ----------
-        mode:
-            Whether to compute function values or residuals.
-        Returns
-        -------
-        flag:
-            Boolean indicating whether mode is supported
-        """
-        raise NotImplementedError()
+        if (
```

Comment:
I was wondering about that in another PR but shouldn't we implement something that checks that the `sensi_orders` are even valid? right now if we would call `check_sensi_orders(sensi_orders=(3,))` it would return `True`.  How about adding an `if(max(sensi_orders)>2): raise ValueError`?

Reply:
would make sense